cpg_utils/cromwell.py

```diff
@@ -429,21 +445,38 @@ def _get_wait_interval(
     status_reported = False
     subprocess.check_output(GCLOUD_ACTIVATE_AUTH_BASE)  # noqa: S603
     cromwell_workflow_root = f'{CROMWELL_URL}/api/workflows/v1/{workflow_id}'
+    abort_url = f'{cromwell_workflow_root}/abort'
     metadata_url = f'{cromwell_workflow_root}/metadata'
     outputs_url = f'{cromwell_workflow_root}/outputs'
     status_url = f'{cromwell_workflow_root}/status'
     _remaining_exceptions = max_sequential_exception_count
     start = datetime.now()
+    cumulative_wait: int = 0
 
     while True:
+        # kill the workflow if the maximum wait has elapsed
+        if time_limit_seconds and cumulative_wait > time_limit_seconds:
+            # time to die, Mr. Cromwell
+            auth_header = {'Authorization': f'Bearer {_get_cromwell_oauth_token()}'}
+            r = requests.post(abort_url, headers=auth_header, timeout=10)
+            if not r.ok:
+                logger.info(f"You couldn't even die right, {workflow_id}")
```

Comment:
I don't expect this to reach main, but I'm really not sure how to handle unkillable eternal jobs 💸

Reply:
Should probably wrap this in some retryability

Reply:
A failure here will fail over to the next loop, where the abort will be retried. That will indirectly provide `max_sequential_exception_count` abort attempts before the watcher errors out. Happy to leave handling there, or hang here and retry the abort in a separate loop?